src/main/java/de/dennisguse/opentracks/io/file/exporter/KmlTrackWriter.java

```diff
@@ -345,7 +347,7 @@ private void writePhotoOverlay(String name, String category, String description,
             writeCategory(category);
 
             if (exportPhotos) {
-                printWriter.println("<Icon><href>" + Uri.decode(photoUrl) + "</href></Icon>");
```

Comment:
I know it was not handled before, but what happens if the URL is invalid?

Reply:
If it's not valid then Marker not show the photo (obviously) because the importer expect that the "last path segment" was valid (we would have a broken link).

I don't know how importer can check it because in the importing process firstly it creates the URI and later copy images to internal memory.

The same for the exporter: firstly creates the URL and then copy the files inside "images".

Reply:
So, we should first the get all files and the parse the KML, right?
Then we can copy the used files into the app-external directory.

For now, I guess, we should just apply a your fix now (then release v3.6) and then rework this properly afterwards.
About the writer: I would prefer to be careful as we never now what is actually in the database (and what some previous OpenTracks version actually did).
@rgmf What do you think?

Reply:
> For now, I guess, we should just apply a your fix now (then release v3.6) and then rework this > properly afterwards.

Ok @dennisguse 

> So, we should first the get all files and the parse the KML, right?
> Then we can copy the used files into the app-external directory.

Checking that URLs are valid ones need more time.

Doing so in the **importer** seems feasible.

But in the exporter seems more difficult because it writes the kml file first and if it success then write images. If you write images firstly and then you write kml file then you have to delete images if kml file couldn't be written. Anyway I have to deep on that.

> About the writer: I would prefer to be careful as we never now what is actually in the database (and what some previous OpenTracks version actually did).

Writer get track, trackpoint and waypoint from database and write them out to the kml file. The only transformation I do is the photoUrl value. In the database is a file:// or a content:// and in the kml file will be a relative path with the same name.

Export works with file:// and content://

Import now generates only content://

Why do you think we should careful with the writer?

Reply:
About the exporter: in the future, we might change photoUrl to just contain the actual filename and not storing the whole content-URL.
And existing installations of OpenTracks might contain some weird things in their databases - there weren't to many changes regarding photoUrl. 
I guess the biggest one was migration to Storage Access Framework.
So, basically we should keep it in mind to not rely too much on the current implementation, but also expects some quirks introduced by previous versions.

Reply:
Ok, tomorrow I'll open an Issue to write down all these changes and I'm going to work on it.

Reply:
And there is another change that affected photoUrl (use of a FileProvider to get images from the camera):
https://github.com/OpenTracksApp/OpenTracks/commit/828a19aea182672374534368a8c09a439c0167f6

It was released with the very first OpenTracks version (v3.0.0).